bin/main.ml

```diff
@@ -901,6 +902,56 @@ let subst =
          )
   , Term.info "subst" ~doc ~man)
 
+let utop =
+  let doc = "Load library in utop" in
+  let man = [ (* TODO *) ] in
+  let go common dir ctx_name args =
+    let utop_target = dir |> Path.of_string |> Utop.utop_exe |> Path.to_string in
+    set_common common ~targets:[utop_target];
+    let log = Log.create () in
+    let (build_system, context, utop_path) =
+      (Main.setup ~log common >>= fun setup ->
+       let context =
+         match ctx_name, setup.contexts with
+         | None, [c] -> c
+         | None, [] -> die "No contexts discovered"
+         | None, ctxs ->
+           die "Specify a --context argument to disambiguate contexts.\
+                @.Available contexts: %s"
+             (ctxs
+             |> List.map ~f:(fun c -> c.Context.name)
+             |> String.concat ~sep:",")
+         | Some name, ctxs ->
+           begin match List.find ctxs ~f:(fun ctx -> ctx.Context.name = name) with
+           | None -> die "Context %s doesn't exist" name
+           | Some c -> c
+           end in
+       let target =
+         match resolve_targets ~log common setup [utop_target] with
+         | [] -> die "no libraries defined in %s" dir
+         | [target] -> target
+         | targets ->
+           match List.find targets ~f:(Path.is_descendant
```

Comment:
What about calling `resolve_targets` with `{setup with contexts = [context] }`? This will ensure that it only returns the executable for this context